src/Faithlife.Tracing/AspNet/AspNetTracing.cs

```diff
@@ -0,0 +1,108 @@
+#if NET461
+using System;
+using System.Threading;
+using System.Web;
+
+namespace Faithlife.Tracing.AspNet
+{
+	internal static class AspNetTracing
+	{
+		public static bool Initialize(HttpApplication application, TracingInitializationInfo info)
+		{
+			var firstTimeInitialization = false;
+
+			if (s_serviceName == null)
+			{
+				lock (s_lock)
+				{
+					if (s_serviceName == null)
+					{
+						if (info == null)
+							throw new ArgumentNullException(nameof(info));
+						if (info.ServiceName == null)
+							throw new ArgumentOutOfRangeException(nameof(info), "info.ServiceName must be set");
+						if (info.CreateTraceFactory == null)
+							throw new ArgumentOutOfRangeException(nameof(info), "info.CreateTraceFactory must be set");
+
+						var samplingRate = info.SamplingRate.GetValueOrDefault(1);
+						SamplingRate =
+							samplingRate < 0 ? throw new ArgumentOutOfRangeException(nameof(info), "info.SamplingRate cannot be negative") :
+							samplingRate > 1 ? throw new ArgumentOutOfRangeException(nameof(info), "info.SamplingRate must be between 0.0 and 1.0") :
+							(int) Math.Round(samplingRate * c_samplingPrecision);
+						TraceFactory = info.CreateTraceFactory();
+						s_serviceName = info.ServiceName;
+
+						firstTimeInitialization = true;
+					}
+				}
+			}
+
+			application.BeginRequest += BeginRequest;
+			application.EndRequest += EndRequest;
+			return firstTimeInitialization;
+		}
+
+		public static void BeginRequest(object sender, EventArgs e)
+		{
+			var httpContext = HttpContext.Current;
+			var headers = httpContext.Request.Headers;
+			var b3Sampled = headers[TracingUtility.B3SampledHeaderName];
+			if (b3Sampled == "0")
```

Comment:
I still don't understand why this isn't `b3Sampled != "1"`.

Reply:
Only `"0"` suppresses tracing.

Reply:
Ah, if it is neither `0` nor `1`, it traces according to the sampling rate; I didn't look closely enough.